apis/v1alpha2/gateway_types.go

```diff
@@ -672,19 +663,27 @@ const (
 
 // ListenerStatus is the status associated with a Listener.
 type ListenerStatus struct {
-	// Port is the unique Listener port value for which this message is
-	// reporting the status.
-	Port PortNumber `json:"port"`
-
-	// Protocol is the Listener protocol value for which this message is
-	// reporting the status.
-	Protocol ProtocolType `json:"protocol"`
+	// Name is the name of the Listener.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +optional
+	Name *string `json:"name,omitempty"`
 
-	// Hostname is the Listener hostname value for which this message is
-	// reporting the status.
+	// SupportedKinds is the list indicating the Kinds supported by this
+	// listener. When this is not specified on the Listener, this MUST represent
+	// the kinds an implementation supports for the specified protocol. When
+	// there are kinds specified on the Listener, this MUST represent the
+	// intersection of those kinds and the kinds supported by the implementation
+	// for the specified protocol.
 	//
+	// +kubebuilder:validation:MaxItems=10
 	// +optional
-	Hostname *Hostname `json:"hostname,omitempty"`
+	SupportedKinds []RouteGroupKind `json:"supportedKinds,omitempty"`
+
+	// AttachedRoutes represents the total number of Routes that have been
+	// successfully attached to this Listener.
+	AttachedRoutes int32 `json:"attachedRoutes"`
```

Comment:
Usually I think of upper limits as being useful to help limit the max size/complexity of a resource. In this case I don't think a limit would help with either of those. I think it could be reasonable for implementations on their own to limit the number of attached routes they'd support per Listener, but I'm not sure there's as much value for us limiting this value here.